Documentation/ApiOverview/FlexForms/Index.rst

```diff
@@ -348,52 +348,112 @@ Extbase controller.
-are available to work with Flexforms from within PHP:
+You can use the :php:`FlexFormService` to read the content of a FlexForm field:
 
-In order to convert a Flexform to a PHP array, the :php:`xml2array` method can
-be used:
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Controller/NonExtbaseController
 
-.. code-block:: php
+    use TYPO3\CMS\Core\Service\FlexFormService;
+    use TYPO3\CMS\Core\Utility\GeneralUtility;
 
-   $flexFormArray = \TYPO3\CMS\Core\Utility\GeneralUtility::xml2array($flexFormString);
+    // Inject FlexFormService
+    public function __construct(
+        protected readonly FlexFormService $flexFormService,
+    ) {
+    }
 
+    private function loadFlexForm(): void
+    {
+        $this->flexFormData = $this->flexFormService
+            ->convertFlexFormContentToArray($this->cObj->data['pi_flexform']);
+    }
 
-In order to convert an PHP array into an Flexform, the :php`flexArray2Xml`
-method can be used:
+The resulting array would look this:
+
+..  code-block:: php
+    :caption: Example output
+
+    $output = $this->flexFormService
+            ->convertFlexFormContentToArray($this->cObj->data['pi_flexform']);
+
+    $looksLike = [
+        'settings' => [
+            'singlePid' => 25,
+            'listPid' => 380,
+        ],
+    ];
+
+While the result of :php:`GeneralUtility::xml2array()` might look like:
 
 .. code-block:: php
 
-   $flexFormTools = new \TYPO3\CMS\Core\Configuration\FlexForm\FlexFormTools();
-   $flexFormString = $flexFormTools->flexArray2Xml($flexFormArray, true);
+    $output = GeneralUtility::xml2array($this->cObj->data['pi_flexform'])
+
+    $looksLike = [
+        'data' =>
+            [
+                'sDEF' =>
+                    [
+                        'lDEF' =>
+                            [
+                                'settings.singlePid' =>['vDEF' => '4',],
+                                'settings.listPid' =>['vDEF' => '',],
+                            ],
+                    ],
+            ],
+    ];
 
 
-.. index:: pair: Flexforms; TypoScript
+..  index::
+    FlexForms; Modify in PHP
+    FlexForms; FlexFormTools
+.. _modify-flexforms-php:
+
+How to modify FlexForms from PHP
+--------------------------------
+
+Some situation make it necessary to modify FlexForms via PHP.
+
+In order to convert a FlexForm to a PHP array, preserving the structure,
+the :php:`xml2array` method in :php:`GeneralUtility`  can be used to read
+the FlexForm data, then the :php:`FlexFormTools` can be used to write back the
+changes.
+
+..  code-block:: php
+
+    use \TYPO3\CMS\Core\Utility\GeneralUtility;
+    use \TYPO3\CMS\Core\Configuration\FlexForm\FlexFormTools;
+
+    $flexFormArray = GeneralUtility::xml2array($flexFormString);
+    $this->doSomething($flexFormArray);
+
+    $flexFormTools = new FlexFormTools();
+    $flexFormString = $flexFormTools->flexArray2Xml($flexFormArray, true);
```

Comment:
```suggestion
    $flexFormString = $flexFormTools->flexArray2Xml($changedFlexFormArray, addPrologue: true);
```

I added also the named argument "addPrologue", so it is clear, what true means. Had to look up the method to get it. Or, if you don't like it, add a comment what the second argument means.